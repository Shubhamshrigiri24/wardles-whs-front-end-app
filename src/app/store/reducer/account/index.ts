import { handleActions } from "redux-actions";
import { AnyAction, combineReducers } from "redux";

import { ACTION_STATUSES, TActionStatus } from "app/store/types";

import { State as AccountState, TAccount, AccountActionTypes } from "./types";
import { ACCOUNT_ACTIONS } from "app/store/reducer/account/constants";

const defaultAccountState: AccountState = {
  data: null,
  status: ACTION_STATUSES.INITIAL,
};

const customerActions = handleActions(
  {
    [ACCOUNT_ACTIONS.SET_ACCOUNT]: (
      _state: TAccount | null,
      action: AccountActionTypes
    ) => {
      const a = action as AnyAction;
      const { payload } = a;

      return payload;
    },
  },
  defaultAccountState.data
);

const statusActions = handleActions(
  {
    [ACCOUNT_ACTIONS.SET_ACCOUNT_STATUS]: (
      _state: TActionStatus,
      action: AccountActionTypes
    ) => {
      const a = action as AnyAction;
      const { payload } = a;

      return payload;
    },
  },
  defaultAccountState.status
);

const reducer = combineReducers({
  data: customerActions,
  status: statusActions,
});

export default reducer;
