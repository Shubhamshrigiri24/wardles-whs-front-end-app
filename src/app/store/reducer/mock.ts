import faker from "faker/locale/en_GB";
import { CustomerDetails, DeliveryDetails } from "./checkout/types";
import { PostcodeLookupAddress } from "./shared";
import {
  OnlineQuestion,
  Pack,
  Product,
  SuccessfulConsultationResponse,
} from "./online/types";

export function makeCustomerDetails(
  override?: Partial<CustomerDetails>
): CustomerDetails {
  return {
    firstName: "firstName",
    lastName: "lastName",
    email: "a@b.com",
    postcode: "M19 2UL",
    line1: "line1",
    line2: "line2",
    city: "city",
    phone: "01189998819991197253",
    surgeryName: "surgeryName",
    surgeryAddress: "surgeryAddress",
    allowMarketing: true,
    ...override,
  };
}

export function makeDeliveryDetails(): DeliveryDetails {
  return {
    title: "Mrs",
    firstName: faker.name.firstName(),
    lastName: faker.name.lastName(),
    postcode: faker.address.zipCode(),
    line1: faker.address.streetName(),
    line2: faker.address.streetName(),
    city: faker.address.city(),
  };
}

export function makePostCodeLookUpAddresses(): PostcodeLookupAddress[] {
  return [
    {
      line1: "64 Broom Avenue",
      line2: "",
      city: "Manchester",
      country: "England",
      postcode: "M19 2UL",
    },
    {
      line1: "66 Broom Avenue",
      line2: "",
      city: "Manchester",
      country: "England",
      postcode: "M19 2UL",
    },
    {
      line1: "68 Broom Avenue",
      line2: "",
      city: "Manchester",
      country: "England",
      postcode: "M19 2UL",
    },
  ];
}

export function makeSuccessfulConsultationResponse(
  id: string = "default",
  override?: Partial<SuccessfulConsultationResponse>
): SuccessfulConsultationResponse {
  return {
    answer: `answer-${id}`,
    answerString: `answerString-${id}`,
    initialQuestion: `question-${id}`,
    kind: `trepetto-input-type-${id}`,
    question: `question-${id}`,
    type: "string",
    ...override,
  };
}

export function makeOnlineQuestion(number: number = 1): OnlineQuestion {
  return {
    number,
    question: `question-${number}`,
    answer: `answer-${number}`,
    type: "string",
  };
}

export function makeOnlineQuestions(): OnlineQuestion[] {
  return [makeOnlineQuestion(1), makeOnlineQuestion(2), makeOnlineQuestion(3)];
}

export function makeProduct(
  id: string = "default",
  override?: {
    id?: string;
    name?: string;
    service?: string;
    packs?: Pack[];
    variant?: string;
  }
): Product {
  return {
    id,
    name: `name-${id}`,
    service: `service-${id}`,
    packs: [
      makePack(`${id}-pack-1`),
      makePack(`${id}-pack-2`),
      makePack(`${id}-pack-3`),
    ],
    variant: "",
    ...override,
  };
}

export function makePack(
  id: string = "default",
  override?: {
    sku?: string;
    name?: string;
    label?: string;
    price?: number;
    pricePerUnit?: number;
    itemsPerPack?: number;
  }
): Pack {
  return {
    sku: `sku-${id}`,
    name: `name-${id}`,
    label: `label-${id}`,
    basePrice: 1,
    price: 1,
    pricePerUnit: 1,
    itemsPerPack: 1,
    discounts: [],
    ...override,
  };
}

export const customerDetails: CustomerDetails = {
  firstName: "FIRST_NAME",
  lastName: "LAST_NAME",
  email: "EMAIL",
  postcode: "POSTCODE",
  line1: "LINE_1",
  line2: "LINE_2",
  city: "CITY",
  phone: "PHONE",
  surgeryName: "SURGERY_NAME",
  surgeryAddress: "SURGERY_ADDRESS",
  allowMarketing: true,
};

export const deliveryDetails: DeliveryDetails = {
  title: "Mrs",
  firstName: "FIRST_NAME",
  lastName: "LAST_NAME",
  postcode: "POSTCODE",
  line1: "LINE_1",
  line2: "LINE_2",
  city: "CITY",
};
