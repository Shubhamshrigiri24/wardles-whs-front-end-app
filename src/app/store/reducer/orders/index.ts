import { handleActions } from "redux-actions";
import { AnyAction, Reducer, combineReducers } from "redux";

import { ACTION_STATUSES, TActionStatus } from "app/store/types";
import { State as OrdersState, TOrders } from "./types";
import { ORDERS_ACTIONS } from "app/store/reducer/orders/constants";

const defaultOrdersState: OrdersState = {
  data: [],
  status: ACTION_STATUSES.INITIAL,
};

const ordersDataActions = handleActions(
  {
    [ORDERS_ACTIONS.SET_ORDERS]: (_state: TOrders, action: AnyAction) => {
      return action.payload;
    },
    [ORDERS_ACTIONS.SET_ORDER]: (_state: TOrders | null, action: AnyAction) => {
      const a = action as AnyAction;
      const { payload } = a;
      const newOrdersArray: TOrders = [...(_state || [])];

      newOrdersArray.push(payload.order);

      return newOrdersArray;
    },
  },
  defaultOrdersState.data
);

const statusActions = handleActions(
  {
    [ORDERS_ACTIONS.SET_ORDERS_STATUS]: (
      _state: TActionStatus,
      action: AnyAction
    ) => {
      return action.payload;
    },
  },
  defaultOrdersState.status
);

const reducer: Reducer<OrdersState, any> = combineReducers({
  data: ordersDataActions,
  status: statusActions,
});

export default reducer;
