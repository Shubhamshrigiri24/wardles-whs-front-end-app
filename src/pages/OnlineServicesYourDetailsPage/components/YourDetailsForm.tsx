import React, { useCallback, useEffect, useState } from "react";
import {
  Typography,
  Button,
  TextField,
  Theme,
  makeStyles,
  FormControl,
  InputLabel,
  Select,
  MenuItem,
  Container,
  Card,
  Checkbox,
  FormHelperText,
} from "@material-ui/core";
import { useForm } from "react-hook-form";
import { mainSchema } from "./schemas";
import { CustomerDetails } from "app/store/reducer/checkout/types";
import { emptyCustomerDetails } from "app/store/reducer/checkout/constants";
import { PostcodeLookupAddress } from "../../../app/store/reducer/shared";
import AddressLookUp from "../../../components/AddressLookUp";

const useStyles = makeStyles((theme: Theme) => ({
  form: {
    padding: theme.breakpoints.down("sm") ? "0px" : theme.spacing(2),
    border: `1px solid ${theme.palette.divider}`,
  },
  text: {
    color: "#445461",
    margin: theme.spacing(3, 0),
  },
  marginInput: {
    margin: theme.spacing(2, 0),
  },
  marginBottom: {
    marginBottom: theme.spacing(3),
  },
  paddingBottom: {
    paddingBottom: theme.spacing(2),
  },
  marginTop: {
    marginTop: theme.spacing(4),
  },
  inputMarginSmall: {
    marginTop: theme.spacing(2),
    width: "100%",
  },
  input: {
    marginTop: theme.spacing(3),
    width: "100%",
  },
  pageMargin: {
    marginTop: theme.spacing(3),
    marginBottom: theme.spacing(15),
  },
  card: {
    width: theme.breakpoints.down("sm") ? "100%" : "580px",
    border: "none",
    background: "none",
  },
  checkboxText: {
    color: "#65727D",
    marginBottom: theme.spacing(3),
  },
  textLink: {
    textDecoration: "none",
    color: "#0061F2",
  },
  checkboxContainer: {
    display: "flex",
    marginTop: theme.spacing(5),
  },
  titleSelect: {
    width: "170px",
  },
  hidden: { display: "none" },
  nextButton: {
    width: "100%",
    backgroundColor: "#233645",
    marginTop: theme.spacing(3),
  },
}));

export interface YourDetailsFormProps {
  onSubmit(data: CustomerDetails): void;
  fetchAddressesByPostcode(postcode: string): void;
  postCodeLookUpAddresses: PostcodeLookupAddress[];
}

interface YourDetailsFormData {
  title: string;
  firstName: string;
  lastName: string;
  email: string;
  phone: string;
  line1: string;
  line2: string;
  city: string;
  postcode: string;
}

const YourDetailsForm: React.FC<YourDetailsFormProps> = (props) => {
  const { onSubmit, postCodeLookUpAddresses, fetchAddressesByPostcode } = props;
  const classes = useStyles();
  // eslint-disable-next-line @typescript-eslint/naming-convention
  const [allowMarketing, setAllowMarketing] = useState(false);
  const {
    register,
    handleSubmit,
    errors,
    reset,
    setValue,
    triggerValidation,
    getValues,
  } = useForm({
    validationSchema: mainSchema,
    defaultValues: emptyCustomerDetails,
    submitFocusError: true,
  });

  useEffect(() => {
    reset(emptyCustomerDetails);
  }, [reset]);

  const handleTitleChange = useCallback(
    (
      event: React.ChangeEvent<{
        name?: string;
        value: unknown;
      }>
    ) => {
      const titleValue = event.target.value as string;
      setValue([{ title: titleValue }]);
      triggerValidation("title");
    },
    [setValue, triggerValidation]
  );

  const setAddressFromAddressLookUp = useCallback(
    (address: PostcodeLookupAddress) => {
      setValue([
        { postcode: address.postcode },
        { line1: address.line1 },
        { line2: address.line2 },
        { city: address.city },
      ]);
      triggerValidation("postcode");
      triggerValidation("line1");
      triggerValidation("line2");
      triggerValidation("city");
    },
    [setValue, triggerValidation]
  );

  const YourDetailsFormOnSubmit = useCallback(
    (data: YourDetailsFormData) => {
      onSubmit({ ...data, allowMarketing });
    },
    [onSubmit, allowMarketing]
  );

  const checkReceiveOffers = useCallback(
    (event: any) => {
      setAllowMarketing(event.target.checked);
    },
    [setAllowMarketing]
  );

  return (
    <Container maxWidth={"sm"} className={classes.pageMargin}>
      <Card className={classes.card}>
        <Typography variant={"h3"}>{"Your details"}</Typography>
        <form
          id={"YourDetails-details-form"}
          noValidate
          autoComplete={"off"}
          onSubmit={handleSubmit(YourDetailsFormOnSubmit)}
        >
          <div className={classes.paddingBottom}>
            <FormControl
              variant={"outlined"}
              className={classes.input}
              error={!!errors.title && !!errors.title.message}
            >
              <InputLabel
                htmlFor={"outlined-title-select"}
                style={{ backgroundColor: "white" }}
              >
                Title
              </InputLabel>
              <Select
                className={classes.titleSelect}
                key={"YourDetailsForm/title-select"}
                value={getValues("title")}
                onChange={handleTitleChange}
                inputRef={register}
                inputProps={{
                  name: "title",
                  "data-testid": "YourDetailsForm/title-select",
                  id: "outlined-title-select",
                }}
              >
                {["Mr", "Mrs", "Miss", "Ms", "Mx", "Sir", "Dr"].map(
                  (title, index) => {
                    return (
                      <MenuItem key={`${title}-${index}`} value={title}>
                        {title}
                      </MenuItem>
                    );
                  }
                )}
              </Select>
              {errors.title && errors.title.message && (
                <FormHelperText>{errors.title.message}</FormHelperText>
              )}
            </FormControl>
            <TextField
              name={"title"}
              inputRef={register}
              className={classes.hidden}
            />
          </div>

          <TextField
            variant={"outlined"}
            fullWidth
            label={"First name"}
            name={"firstName"}
            error={!!errors.firstName}
            helperText={errors.firstName && errors.firstName.message}
            inputRef={register}
            InputLabelProps={{
              shrink: true,
            }}
            inputProps={{
              "data-testid": "YourDetailsForm/firstName",
            }}
          />

          <TextField
            className={classes.inputMarginSmall}
            label={"Last name"}
            variant={"outlined"}
            fullWidth
            name={"lastName"}
            error={!!errors.lastName}
            helperText={errors.lastName && errors.lastName.message}
            inputRef={register}
            InputLabelProps={{
              shrink: true,
            }}
            inputProps={{
              "data-testid": "YourDetailsForm/lastName",
            }}
          />

          <TextField
            className={classes.inputMarginSmall}
            label={"Email"}
            variant={"outlined"}
            fullWidth
            name={"email"}
            error={!!errors.email}
            helperText={errors.email && errors.email.message}
            inputRef={register}
            InputLabelProps={{
              shrink: true,
            }}
            inputProps={{
              "data-testid": "YourDetailsForm/email",
            }}
          />

          <TextField
            className={classes.marginInput}
            label={"Phone number"}
            variant={"outlined"}
            fullWidth
            name={"phone"}
            error={!!errors.phone}
            helperText={errors.phone && errors.phone.message}
            inputRef={register}
            InputLabelProps={{
              shrink: true,
            }}
            inputProps={{
              "data-testid": "YourDetailsForm/phone",
            }}
          />

          <AddressLookUp
            postCodeLookUpAddresses={postCodeLookUpAddresses}
            fetchAddressesByPostcode={fetchAddressesByPostcode}
            setAddress={setAddressFromAddressLookUp}
            error={errors?.postcode?.message?.toString() || ""}
            inputRef={register}
          />

          <TextField
            className={classes.input}
            label={"Address line 1"}
            variant={"outlined"}
            fullWidth
            name={"line1"}
            error={!!errors.line1}
            helperText={errors.line1 && errors.line1.message}
            inputRef={register}
            InputLabelProps={{
              shrink: true,
            }}
            inputProps={{
              "data-testid": "YourDetailsForm/line1",
            }}
          />
          <TextField
            className={classes.input}
            label={"Address line 2 (optional)"}
            variant={"outlined"}
            fullWidth
            name={"line2"}
            error={!!errors.line2}
            helperText={errors.line2 && errors.line2.message}
            inputRef={register}
            InputLabelProps={{
              shrink: true,
            }}
            inputProps={{
              "data-testid": "YourDetailsForm/line2",
            }}
          />
          <TextField
            className={classes.input}
            label={"City"}
            variant={"outlined"}
            fullWidth
            name={"city"}
            error={!!errors.city}
            helperText={errors.city && errors.city.message}
            inputRef={register}
            InputLabelProps={{
              shrink: true,
            }}
            inputProps={{
              "data-testid": "YourDetailsForm/city",
            }}
          />
          <div className={classes.checkboxContainer}>
            <div>
              <Checkbox
                onChange={checkReceiveOffers}
                style={{
                  paddingLeft: "0px",
                }}
                inputProps={{
                  // @ts-ignore
                  "data-testid": "YourDetailsForm/checkbox",
                }}
              />
            </div>
            <div>
              <Typography variant={"caption"} className={classes.checkboxText}>
                Optional: I would like to receive health advice and offers from
                Well by email. (You can change your marketing preferences at any
                time)
              </Typography>
            </div>
          </div>

          <Button
            type={"submit"}
            color={"primary"}
            variant={"contained"}
            data-testid={"YourDetailsForm/next-button"}
            className={classes.nextButton}
          >
            Next
          </Button>
        </form>
        <div className={classes.input}>
          <Typography variant={"caption"}>
            By entering your details and payment information, you are agreeing
            to our{" "}
            <a
              className={classes.textLink}
              href={
                "https://www.well.co.uk/about-us/policies/shop-terms-and-conditions"
              }
              target={"_blank"}
              rel={"noopener noreferrer"}
            >
              Terms and Conditions
            </a>
            .
          </Typography>
        </div>
      </Card>
    </Container>
  );
};

export default YourDetailsForm;
