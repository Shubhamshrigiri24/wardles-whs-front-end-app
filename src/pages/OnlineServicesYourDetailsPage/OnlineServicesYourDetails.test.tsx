import React from "react";
import { render, fireEvent, act } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import YourDetailsForm, {
  YourDetailsFormProps,
} from "./components/YourDetailsForm";
import AddressLookUp, {
  AddressLookUpProps,
} from "../../components/AddressLookUp";
import { makePostCodeLookUpAddresses } from "app/store/reducer/mock";

const mockHistoryPush = jest.fn();
jest.mock("react-router-dom", () => ({
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}));
beforeEach(() => {
  mockHistoryPush.mockReset();
});

describe("<YourDetailsForm />", () => {
  it("should match the snapshot", () => {
    const props = makeProps();
    const { container } = render(<YourDetailsForm {...props} />);
    expect(container).toMatchSnapshot();
  });
  it("should call onSubmit on form submit with YourDetails details", async () => {
    const props = makeProps();
    const { getByTestId } = render(<YourDetailsForm {...props} />);

    await act(async () => {
      fireEvent.change(getByTestId("YourDetailsForm/firstName"), {
        target: { value: "first-name" },
      });
    });
    await act(async () => {
      fireEvent.change(getByTestId("YourDetailsForm/lastName"), {
        target: { value: "last-name" },
      });
    });
    await act(async () => {
      fireEvent.change(getByTestId("YourDetailsForm/email"), {
        target: { value: "hello@email.com" },
      });
    });
    await act(async () => {
      fireEvent.change(getByTestId("YourDetailsForm/phone"), {
        target: { value: "07777777777" },
      });
    });
    await act(async () => {
      fireEvent.change(getByTestId("YourDetailsForm/line1"), {
        target: { value: "address line 1" },
      });
    });
    await act(async () => {
      fireEvent.change(getByTestId("YourDetailsForm/city"), {
        target: { value: "YourDetails city" },
      });
    });

    const addressLookupProps = makeAddressLookupProps();

    const { getAllByTestId: getAddressLookUpTestById } = render(
      <AddressLookUp {...addressLookupProps} />
    );
    await act(async () => {
      getAddressLookUpTestById("postcode").forEach((test) => {
        fireEvent.change(test, {
          target: { value: "A1 1AA" },
        });
      });
    });
    await act(async () => {
      fireEvent.change(getByTestId("YourDetailsForm/title-select"), {
        target: {
          value: "Miss",
        },
      });
    });
    await act(async () => {
      fireEvent.click(getByTestId("YourDetailsForm/checkbox"));
    });
    await act(async () => {
      fireEvent.click(getByTestId("YourDetailsForm/next-button"));
    });

    expect(props.onSubmit).toBeCalledWith({
      title: "Miss",
      firstName: "first-name",
      lastName: "last-name",
      email: "hello@email.com",
      phone: "07777777777",
      postcode: "A1 1AA",
      line1: "address line 1",
      line2: "",
      city: "YourDetails city",
      allowMarketing: true,
    });
  });
});

function makeProps(v?: Partial<YourDetailsFormProps>): YourDetailsFormProps {
  return {
    onSubmit: jest.fn(),
    fetchAddressesByPostcode: jest.fn(),
    postCodeLookUpAddresses: makePostCodeLookUpAddresses(),
  };
}

function makeAddressLookupProps(
  v?: Partial<AddressLookUpProps>
): AddressLookUpProps {
  return {
    fetchAddressesByPostcode: jest.fn(),
    postCodeLookUpAddresses: makePostCodeLookUpAddresses(),
    setAddress: jest.fn(),
    className: "",
    error: "",
    inputRef: "",
  };
}
