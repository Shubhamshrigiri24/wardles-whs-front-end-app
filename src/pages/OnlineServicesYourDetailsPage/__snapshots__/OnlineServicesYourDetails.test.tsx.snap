// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<YourDetailsForm /> should match the snapshot 1`] = `
<div>
  <div
    class="MuiContainer-root makeStyles-pageMargin-9 MuiContainer-maxWidthSm"
  >
    <div
      class="MuiPaper-root MuiCard-root makeStyles-card-10 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3"
      >
        Your details
      </h3>
      <form
        autocomplete="off"
        id="YourDetails-details-form"
        novalidate=""
      >
        <div
          class="makeStyles-paddingBottom-5"
        >
          <div
            class="MuiFormControl-root makeStyles-input-8"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink="false"
              for="outlined-title-select"
              style="background-color: white;"
            >
              Title
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root makeStyles-titleSelect-14 MuiInputBase-formControl"
            >
              <div
                aria-haspopup="listbox"
                aria-labelledby="mui-component-select-title"
                class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
                id="mui-component-select-title"
                role="button"
                tabindex="0"
              >
                <span>
                  ​
                </span>
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput"
                data-testid="YourDetailsForm/title-select"
                id="outlined-title-select"
                name="title"
                tabindex="-1"
                value=""
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-17 MuiOutlinedInput-notchedOutline"
                style="padding-left: 8px;"
              >
                <legend
                  class="PrivateNotchedOutline-legend-18"
                  style="width: 0.01px;"
                >
                  <span>
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root makeStyles-hidden-15"
          >
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input"
                name="title"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined"
            data-shrink="true"
          >
            First name
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input"
              data-testid="YourDetailsForm/firstName"
              name="firstName"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-17 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-19 PrivateNotchedOutline-legendNotched-20"
              >
                <span>
                  First name
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root makeStyles-inputMarginSmall-7 MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined"
            data-shrink="true"
          >
            Last name
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input"
              data-testid="YourDetailsForm/lastName"
              name="lastName"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-17 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-19 PrivateNotchedOutline-legendNotched-20"
              >
                <span>
                  Last name
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root makeStyles-inputMarginSmall-7 MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined"
            data-shrink="true"
          >
            Email
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input"
              data-testid="YourDetailsForm/email"
              name="email"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-17 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-19 PrivateNotchedOutline-legendNotched-20"
              >
                <span>
                  Email
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root makeStyles-marginInput-3 MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined"
            data-shrink="true"
          >
            Phone number
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input"
              data-testid="YourDetailsForm/phone"
              name="phone"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-17 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-19 PrivateNotchedOutline-legendNotched-20"
              >
                <span>
                  Phone number
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-9"
          >
            <div
              class="MuiFormControl-root MuiTextField-root makeStyles-input-21"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined"
                data-shrink="true"
                for="postcode-input"
                id="postcode-input-label"
              >
                Postcode
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  data-testid="postcode"
                  id="postcode-input"
                  name="postcode"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-17 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-19 PrivateNotchedOutline-legendNotched-20"
                  >
                    <span>
                      Postcode
                    </span>
                  </legend>
                </fieldset>
              </div>
              
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-searchButton-22 Mui-disabled Mui-disabled"
              data-testid="addressLookUp/search-button"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Search
              </span>
            </button>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root makeStyles-input-8 MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined"
            data-shrink="true"
          >
            Address line 1
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input"
              data-testid="YourDetailsForm/line1"
              name="line1"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-17 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-19 PrivateNotchedOutline-legendNotched-20"
              >
                <span>
                  Address line 1
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root makeStyles-input-8 MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined"
            data-shrink="true"
          >
            Address line 2 (optional)
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input"
              data-testid="YourDetailsForm/line2"
              name="line2"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-17 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-19 PrivateNotchedOutline-legendNotched-20"
              >
                <span>
                  Address line 2 (optional)
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root makeStyles-input-8 MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined"
            data-shrink="true"
          >
            City
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input"
              data-testid="YourDetailsForm/city"
              name="city"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-17 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-19 PrivateNotchedOutline-legendNotched-20"
              >
                <span>
                  City
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="makeStyles-checkboxContainer-13"
        >
          <div>
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-24 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
              style="padding-left: 0px;"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-27"
                  data-indeterminate="false"
                  data-testid="YourDetailsForm/checkbox"
                  type="checkbox"
                  value=""
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
          </div>
          <div>
            <span
              class="MuiTypography-root makeStyles-checkboxText-11 MuiTypography-caption"
            >
              Optional: I would like to receive health advice and offers from Well by email. (You can change your marketing preferences at any time)
            </span>
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-nextButton-16 MuiButton-containedPrimary"
          data-testid="YourDetailsForm/next-button"
          tabindex="0"
          type="submit"
        >
          <span
            class="MuiButton-label"
          >
            Next
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </form>
      <div
        class="makeStyles-input-8"
      >
        <span
          class="MuiTypography-root MuiTypography-caption"
        >
          By entering your details and payment information, you are agreeing to our
           
          <a
            class="makeStyles-textLink-12"
            href="https://www.well.co.uk/about-us/policies/shop-terms-and-conditions"
            rel="noopener noreferrer"
            target="_blank"
          >
            Terms and Conditions
          </a>
          .
        </span>
      </div>
    </div>
  </div>
</div>
`;
